dev/release_notes.readme.md

```diff
@@ -0,0 +1,102 @@
+| 10 | Packages | `topic: packages` | PRs related to package handling, or specific to a package (for packages w/o issue tracker) |
+| 11 | Other changes | `release notes: needed` OR `release notes: added` | PRs relevant for the release notes but without a label that fits --in each case, think about assigning such a label or introducing a new one |
+
+- For empty groups of pull requests, the subheading need not appear in `CHANGES.md`.
+
+- If the group "Other changes" is too large then we should decide whether other labels (most likely some `topic: ...` labels) can be used to define useful groups; in this case, a subheading has to be defined for this label.
+  (There will be a script for creating the current `CHANGES.md` section in the `dev` subdirectory of the main GAP directory, the change of the rules should be made in this script.)
+
+- Concerning the ordering, I think that somebody who reads the release notes of a new GAP version will be first interested in highlights, then in bugs which may affect one's computations (wrong results), then in removed functionality (which may break old private code), then in general improvements.
+
+  I think there is no way to satisfy all perspectives of what is most important, in particular concerning the `topic: ...` labels.
+  Note that `CHANGES.md` is just **one** way to present the changes, eventually we should put the source data (pull request number, release notes text, labels) into the GAP distribution, and provide a tool for evaluating the data inside a GAP session.
+
+- For convenience, consistency checks can be executed, via suitable queries (restricted to the merged pull requests that belong to the forthcoming release), for example:
+  - There should be no pull request without the labels `release notes: not needed` and `release notes: use title` AND without release notes text in the body.
+
+- The three sections on changes packages in the forthcomiing release (see above) will appear below the groups of pull requests that are defined by the above labels.
+  (There will be a script that extracts the information from suitable text files related to the package archives.)
+
+
+## How to deal with github pull requests in a release notes friendly way?
+
+The following is intended for those who create or review pull requests.
+
+- If your pull request should not be mentioned in the release notes then add the label `release notes: not needed`, and you are done.
+
+- Otherwise, try to find a short title that describes the pull request in the release notes; use appropriate maarkup in the title (e.g., backquotes for surrounding function names).  In this case, add the label `release notes: use title`.
+
+- Otherwise, make sure that the body of the pull request contains the relevant text below the line `## Text for release notes`.
+  (In the old workflow, the label `release notes: needed` should be added. I think this should be abolished.)
```

Comment:
Instead of "release notes: dont use title" we can also use "release notes:needed" as suggested by Thomas here. Then we only have to delete the "release notes: added" label.

Reply:
I think we need only two labels to define which pull requests appear in the release notes, and what is the text that is shown:
- `release notes: not needed` for pull requests which shall not be mentioned in the release notes,
- `release notes: use title` for pull requests which shall be mentioned in the release notes AND for which the text for the release notes is given by the title.

Then the pull requests *without these labels* are those which shall be mentioned in the release notes AND for which the text for the release notes can be found in the body.

Reply:
Even with the automation, I think it is not realistic to expect that all PRs will always have the right labels and/or a good release not text (be it in the title or label). So I still expect that as we ramp up for a release, one or more dedicated people will go through PRs to "get them into shape" in that regard. For this process, I think it will still be useful to be able to distinguish between "unprocessed" PRs, and those which already have been taken care of (i.e. either have a good title and the "use title" label; or have a good text in the body, and some label indicating that; or are not needed for the release notes)